app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt

```diff
@@ -33,38 +36,44 @@ class SkillsListAdapter(
-        }
+        try {
+            if (skillData.skillName.isEmpty()) {
+                holder.skillName.text = context.getString(R.string.no_skill_name)
+            } else {
+                holder.skillName.text = skillData.skillName
+            }
 
-        if (skillData.author.isEmpty() || skillData.author.equals(DEFAULT_AUTHOR)) {
-            holder.skillAuthorName.text = context.getString(R.string.no_skill_author)
-        } else {
-            holder.skillAuthorName.text = skillData.author
-        }
+            if (skillData.author.isEmpty() || skillData.author.equals(DEFAULT_AUTHOR)) {
+                holder.skillAuthorName.text = context.getString(R.string.no_skill_author)
+            } else {
+                holder.skillAuthorName.text = skillData.author
+            }
 
-        if (skillData.examples.isEmpty() || skillData.examples.contains(DEFAULT_EXAMPLE))
-            holder.skillExample.text = ""
-        else
-            holder.skillExample.text = StringBuilder("\"").append(skillData.examples[0]).append("\"")
+            if (skillData.examples.isEmpty() || skillData.examples.contains(DEFAULT_EXAMPLE))
+                holder.skillExample.text = ""
+            else
+                holder.skillExample.text = StringBuilder("\"").append(skillData.examples[0]).append("\"")
 
-        if (skillData.image.isEmpty()) {
-            holder.skillImage.setImageResource(R.drawable.ic_susi)
-        } else {
-            val imageUrl: String = skillDetails.skillsList.get(position).group.replace(" ", "%20") + "/en/" + skillData.image
-            Picasso.get().load(StringBuilder(imageLink)
-                .append(imageUrl).toString())
-                .fit().centerCrop()
-                .error(R.drawable.ic_susi)
-                .transform(CircleTransform())
-                .into(holder.skillImage)
-        }
-        val stars = skillData.skillRating?.stars
-        if (stars != null) {
-            holder.skillRating.rating = stars.averageStar
-            holder.skillTotalRatings.text = stars.totalStar.toString()
+            if (skillData.image.isEmpty()) {
+                holder.skillImage.setImageResource(R.drawable.ic_susi)
+            } else {
+                val imageUrl: String = skillDetails.skillsList.get(position).group.replace(" ", "%20") + "/en/" + skillData.image
+                Picasso.get().load(StringBuilder(imageLink)
+                        .append(imageUrl).toString())
+                        .fit().centerCrop()
+                        .error(R.drawable.ic_susi)
+                        .transform(CircleTransform())
+                        .into(holder.skillImage)
+            }
+            val stars = skillData.skillRating?.stars
+            if (stars != null) {
+                holder.skillRating.rating = stars.averageStar
+                holder.skillTotalRatings.text = stars.totalStar.toString()
+            }
+        } catch (illegalArgument: IllegalArgumentException) {
```

Comment:
Why is IllegalArgumentException thrown?

Reply:
I was getting an error before due to this, but I think it's working fine without it. So no need for this.